Library/Homebrew/brew.rb

```diff
@@ -32,10 +32,12 @@ class MissingEnvironmentVariables < RuntimeError; end
   raise e if ENV["HOMEBREW_MISSING_ENV_RETRY"]
 
   if ENV["HOMEBREW_DEVELOPER"]
-    $stderr.puts <<~EOS
-      Warning: #{e.message}
-      Retrying with `exec #{ENV["HOMEBREW_BREW_FILE"]}`!
-    EOS
+    Tty.with($stderr) do |stderr|
+      stderr.puts <<~EOS
+        Warning: #{e.message}
```

Comment:
```suggestion
        opoo e
```

Reply:
although: `opoo` may not be defined at this point.

Reply:
Yeah that's what I'm concerned about. In fact even `Tty` seems problematic, as we may not be able to use it unless `require_relative "global"` is successful. Also, since it seems like this particular `e.message` doesn't require formatting, we can just stick to `$stderr.puts`.

Reply:
Let's just stick to using `$stderr.puts` in this file.